Src/Newtonsoft.Json.UnityConverters/Camera/CullingGroupEventConverter.cs

```diff
@@ -0,0 +1,127 @@
+            byte previousDistance = values.GetAsTypeOrDefault<byte>(4);
+
+            int thisStateByte = (byte)(isVisibleByte | (currentDistance & DISTANCE_MASK));
+            int prevStateByte = (byte)(wasVisibleByte | (previousDistance & DISTANCE_MASK));
+
+            var instance = new CullingGroupEvent();
+            TypedReference reference = __makeref(instance);
+
+            _indexField.SetValueDirect(reference, index);
+            _prevStateField.SetValueDirect(reference, prevStateByte);
+            _thisStateField.SetValueDirect(reference, thisStateByte);
+
+            return instance;
+        }
+
+        protected override object[] ReadInstanceValues(CullingGroupEvent instance)
+        {
+            return new object[] {
+                instance.index,
+                instance.isVisible,
+                instance.wasVisible,
+                instance.currentDistance,
+                instance.previousDistance,
+            };
+        }
+
+        protected override object ReadValue(JsonReader reader, int index, JsonSerializer serializer)
+        {
+            return index switch
+            {
+                0 => reader.ReadAsInt32() ?? 0,
+                1 => reader.ReadAsBoolean() ?? false,
+                2 => reader.ReadAsBoolean() ?? false,
+                3 => ReadDistance(reader),
+                4 => ReadDistance(reader),
+
+                _ => throw new ArgumentOutOfRangeException(nameof(index), index, "Only accepts member index in range 0..4")
+            };
+
+            static byte ReadDistance(JsonReader reader)
+            {
+                int value = reader.ReadAsInt32() ?? 0;
+                if (value >= 0x80 || value < 0)
+                {
+                    throw reader.CreateSerializationException($"Overflow in {typeof(CullingGroupEvent).FullName} distance value. Distance must be between 0..127 (inclusive), got {value}");
+                }
+                return (byte)value;
+            }
+        }
+
+        protected override void WriteValue(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            switch (value)
+            {
+            case null:
+                writer.WriteNull();
+                break;
+
+            case byte distance:
+                writer.WriteValue(distance);
+                break;
+
+            case int index:
+                writer.WriteValue(index);
+                break;
+
+            case bool isVisible:
+                writer.WriteValue(isVisible);
+                break;
+
+            default:
+                throw writer.CreateWriterException($"Unexpected type '{value.GetType().Name}' when serializing {typeof(CullingGroupEvent).FullName}");
+            }
+        }
+
+        private static byte AsVisibility(bool isVisible)
+        {
+            return isVisible ? (byte)0x80 : (byte)0;
+        }
+
+        private static byte AsDistance(byte distance)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private method 'AsDistance'.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity.Converters/pullRequest?prid=5420094)